backend/accounts/account.go

```diff
@@ -77,6 +78,9 @@ type Interface interface {
 	SafelloBuy() *safello.Buy
 
 	Notes() *notes.Notes
+	// ProposeTxnote stores a note. The note is is persisted in the notes database upon calling
+	// SendTx(). This function must be called before `SendTx()`.
+	ProposeTxNote(string)
```

Comment:
It's interesting that this method is in the interface but `GetAndClearProposedTxNote` isn't. Maybe `ProposeTxNote` shouldn't be here either? Just feels a bit inconsistent.

Also, what if the the frontend simply makes to calls, send tx and then set note? You wouldn't even need to store a tx proposal note on the backend at all then.

Reply:
"what if the frontend simply makes **two** calls"

Reply:
It's in the interface if it is accessed by handlers.go, the only place where the interface is used.

> Also, what if the the frontend simply makes to calls, send tx and then set note? You wouldn't even need to store a tx proposal note on the backend at all then.

The frontend does not have the txID required to do that. And it seemed more robust this way, then the frontend cannot fuck up.